src/main/scala/is/hail/expr/ir/functions/MathFunctions.scala

```diff
@@ -43,6 +44,16 @@ object MathFunctions extends RegistryFunctions {
 
   def floorDiv(x: Double, y: Double): Double = math.floor(x / y)
 
+  def approxEqual(x: Double, y: Double, tolerance: Double, absolute: Boolean, nanSame: Boolean): Boolean = {
+    x == y || {
+      (if (absolute)
+        math.abs(x - y) <= tolerance
+      else
+        D_==(x, y, tolerance)) ||
+        (nanSame && x.isNaN && y.isNaN)
```

Comment:
This is the same formatting that had me confused in `valuesSimilar`. The logical grouping is completely different than what the indentation suggests. How about factoring it out something like
```scala
val withinTol =
  if (absolute)
    math.abs(x - y) <= tolerance
  else
    D_==(x, y, tolerance)

x == y || withinTol || (nanSame && x.isNaN && y.isNan)
```